### PR TITLE
[elm/en] basics of Elm in English 

### PR DESCRIPTION
**Pull request by GlenDC**
I've been using Elm now since about a month. And I quite like the language. as learnxinyminutes has often helped me for other languages I thought I could do something back by providing similar help to people who want to learn more about Elm. So here you can find the first version of just that.

### CODE REVIEW

File: elm.html.markdown [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+filename: learnelm.elm
+---
+
+[Elm](http://elm-lang.org) is a pure functional programming language for writing web-apps in a safe and intuitive way. Elm uses reactivity on top of the functional style, to provide a bridge between the app and the outside world, without giving up the advantages it has from being pure and functional.
+
+You can experiment with the language via [http://elm-lang.org/try](http://elm-lang.org/try). The [syntax guide](http://elm-lang.org/docs/syntax) is a good start point to learn the language, and a lot of this guide is based on that one. If you're already familiar with Javascript, which is the language elm compiles to, you might want to read [http://elm-lang.org/docs/from-javascript](http://elm-lang.org/docs/from-javascript).
+
+We only cover the basics here. After this you might want to learn more about Signals, which allow you to react on user input and other changes from the outside world. You can learn more about that [here](http://elm-lang.org/guide/reactivity#signals).
+
+```elm
```
Comment (levibostian) [inline elm.html.markdown]: All looks great! Only one problem. elm is not supported in the syntax highlighter library. Look through [this list](https://github.com/adambard/rouge/tree/master/lib/rouge/lexers) of files to see what language is closest to elm and use that one.

**Comment by levibostian** [comment]
Thank you very much! Only this one issue to fix and then we can get this merged in.

**Comment by GlenDC** [comment]
@levibostian I don't think there is a language close enough without bringing confusion in the syntax highlighting. Especially the way Elm comments seems to be quit distinctive.

I might contribute to that library first, to support Elm.

**Comment by vendethiel** [comment]
wouldn't haskell make sense?

File: elm.html.markdown [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+True && not (True || False)
+(2 + 4) * (4^2 - 9)
+"abc" ++ "def"
+
+-- There are no symbols/atoms, as you have in
+-- other languages.
+
+-- Tuples that are stored contiguously in memory.
+(1, 2, 3) -- tuple
```
Comment (vendethiel) [inline elm.html.markdown]: uh?

File: elm.html.markdown [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+-- There is a special function for creating tuples:
```
Comment (vendethiel) [inline elm.html.markdown]: There **are** (plural)?

File: elm.html.markdown [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+(,) 1 2              == (1,2)
+(,,,) 1 True 'a' []  == (1,True,'a',[])
+
+{-----------------------------
+----- Maybe
+-----------------------------}
+
+-- Nil/Null is not possible in Elm, but has a special type instead.
```
Comment (vendethiel) [inline elm.html.markdown]: but **it** has?

File: elm.html.markdown [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+-- A Maybe can help you with optional arguments
```
Comment (vendethiel) [inline elm.html.markdown]: missing a comma here?

File: elm.html.markdown [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+-----------------------------}
+
+-- Elm has Lists as well, which is included as a core module.
+[1,2,3] -- List Int
+
+-- We can access the head and tail of a list as follows:
+List.head [1,2,3] == Just 1
+List.tail [1,2,3] == Just [2,3]
+
+-- Here are four things that are equivalent:
```
Comment (vendethiel) [inline elm.html.markdown]: things -> lists?

File: elm.html.markdown [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+-----------------------------}
+
+-- Where you would use enumerations in other languages
+-- you can use union types in Elm.
+
+type List = Empty | Node Int List
+
+-- The values have to be unique on a per-scope basis
+-- Values can also contain subtypes as you can see above.
+-- Union Types are quite useful for pattern matching
```
Comment (vendethiel) [inline elm.html.markdown]: "for"->"in"?

File: elm.html.markdown [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+
+-- Anonymous functions look very similar to normal functions.
+square = \n -> n ^ 2
+
+-- This can be handy in case you need a specialised function
+-- to be applied to for example a List of elements
+List.map (\n -> n ^ 2) [2, 4, 6] == [4, 16, 36]
+
+-- In order to reduce parentheses usage you can use `|>` and `<|`.
+-- These `Infix Operators` are alias for function application
```
Comment (vendethiel) [inline elm.html.markdown]: alias**es**?

File: elm.html.markdown [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+-- Imagine for example that we want to create & transform a circle:
+
+dot =
+  scale 2 (move (20,20) (filled blue (circle 10)))
+
+-- As you can see, parentheses add up quickly.
+-- Therefore you can and should use the above mentioned `Infix Operators`
+dot' =
+  circle 10 -- creating a circle with a radius of 10
+    |> filled blue -- filling that circle with the color blue
```
Comment (vendethiel) [inline elm.html.markdown]: looks weird without currying mentioned, to be honest

**Comment by geoffliu** [comment]
@vendethiel @GlenDC Any reason to keep this PR open, now that #2054 is merged?

**Comment by vendethiel** [comment]
I don't think so. @GlenDC thanks for your work still :-)!

**Pull request closed without merging**